core/services/log/broadcaster.go

```diff
@@ -401,18 +405,32 @@ func (b *broadcaster) onNewHeads() {
 			keptDepth = 0
 		}
 
-		logs, minBlockNum := b.logPool.getLogsToSend(latestBlockNum)
-
-		if len(logs) > 0 {
-			broadcasts, err := b.orm.FindConsumedLogs(minBlockNum, latestBlockNum)
-			if err != nil {
-				logger.Errorf("LogBroadcaster: Failed to query for log broadcasts, %v", err)
-				return
+		// if all subscribers requested 0 confirmations, we always get and delete all logs from the pool,
+		// without comparing their block numbers to the current head's block number.
+		if b.registrations.lowestNumConfirmations == 0 && b.registrations.highestNumConfirmations == 0 {
```

Comment:
Wouldn't just b.registrations.highestNumConfirmations == 0 be enough here?

Reply:
ah yes actually very good point!